### PR TITLE
Adds a simple Makefile with the most commonly used options

### PR DESCRIPTION
**Pull request by johnSchnake**
This will make it easier for people new to the project to figure
out how to contribute or test changes locally.

We should update some of the documentation accordingly as well.

Signed-off-by: John Schnake <jschnake@vmware.com>

Fixes: #1720

### CODE REVIEW

**Comment by johnSchnake** [comment]
The failure is unrelated to this change and will be fixed immediately.

**Comment by johnSchnake** [comment]
Not changing logic so not going to be blocked by tests, but creating an issue for the failed test.

**Comment by johnSchnake** [comment]
#1750 created regarding the test failure. Unrelated to this PR.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.PHONY: build test update_golden help
+.DEFAULT_GOAL := help
+
+build: ## Builds the Sonobuoy binary for your local OS
+	@echo "Building the sonobuoy binary..."
+	source ./scripts/build_funcs.sh; native
+	@echo "Complete. Use sonobuoy in the pwd (e.g. ./sonobuoy)"
```
Comment (andrewyunt) [inline Makefile:7]: nit: Perhaps a statement that makes more sense would be "Use sonobuoy in the current directory" (pwd stands for print working directory)

Comment (johnSchnake) [reply Makefile:7]: ha, I've always used it interchangeably with the current working directory (cwd) and intuitively assumed it was the 'present working directory'. TIL. 👍

Comment (johnSchnake) [reply Makefile:7]: Looking at this build target I am going to actually swap it out for a simple `go build`. The reasoning is that I realized that using this build_func, just like the others, will reset the sonobuoy versions as if we are cutting a release.

The result will be what you experienced the other day: `sonobuoy run` will lead to setting the image to a non-existant image and you'll get ErrImagePull.

**Review by andrewyunt — APPROVED** [review]

**Pull request merged**